src/main/modules/expense/components/AddExpenseForm.jsx

```diff
@@ -0,0 +1,148 @@
+import { Formik, Form } from "formik";
+import * as Yup from "yup";
+import { FormInputField, FormSelectField } from "./atom/FormFields";
+
+import { FaMoneyBillWave } from "react-icons/fa6";
+import { CgNotes } from "react-icons/cg";
+import { LiaPenAltSolid } from "react-icons/lia";
+import { FaRegCalendarXmark } from "react-icons/fa6";
+
+import { StyledButton } from "../../../../styledComponents";
+import { addCollectionData } from "../../../../service/firebase/expense.service";
+import { COLLECTIONS } from "../../../firebase";
+import { useAuth } from "../../../../hooks/useAuth";
+
+const AddExpenseFormm = () => {
+  const { user } = useAuth();
+  const setDates = (dateData) => {
+    const weekday = [
+      "Sunday",
+      "Monday",
+      "Tuesday",
+      "Wednesday",
+      "Thursday",
+      "Friday",
+      "Saturday",
+    ];
+
+    const date = new Date(dateData);
+    const currentDate = date.getDate();
+    const currentDay = weekday[date.getDay()];
+    const currentMonth = parseInt(date.getMonth()) + 1;
+    const currentYear = date.getFullYear();
+
+    return {
+      currentDate,
+      currentDay,
+      currentMonth,
+      currentYear,
+    };
+  };
+  return (
+    <>
+      <Formik
+        initialValues={{
+          name: "",
+          amount: "",
+          notes: "",
+          date: "",
+          category: "", // added for our select
+        }}
+        validationSchema={Yup.object({
+          name: Yup.string()
+            .max(20, "Must be 20 characters or less")
+            .required("Required"),
+          amount: Yup.number().required("Required"),
+          date: Yup.date()
+            .max(new Date(), 'please check the date you enterd'),
+          notes: Yup.string().max(30, "Must be 30 characters or less"),
+          category: Yup.string()
+            // .oneOf(
```

Comment:
remove unwanted comments, or if the comments are needed add a **// TODO:** comment

Reply:
DONE